### PR TITLE
[Bugfix] New variables to configure account type value filters

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** [#97]

**This PR will result in the following new package version:**

Not a breaking change as most customers rely on the default values for these fields so only a subset should need to apply these changes.

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:** 
- Updated intermediate double entry models that have `account_type` and `account_sub_type` filters with configurable variables, since the type names used in the filter can be adjusted internally by QuickBooks customers. The intermediate models where these variables were introduced included: 

| **Updated model** | **New variables to filter on** |
| ----------| -------------------- |
| [int_quickbooks__bill_payment_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__bill_payment_double_entry) | `quickbooks__accounts_payable_reference` |
| [int_quickbooks__credit_memo_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__credit_memo_double_entry) | `quickbooks__accounts_receivable_reference` |
| [int_quickbooks__deposit_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__deposit_double_entry) | `quickbooks__undeposited_funds_reference` |
| [int_quickbooks__invoice_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__invoice_double_entry) | `quickbooks__sales_of_product_income_reference`, `quickbooks__accounts_receivable_reference` |
| [int_quickbooks__payment_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__payment_double_entry) | `quickbooks__accounts_receivable_reference` |

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [NA] dbt run –vars (if applicable)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [x] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [x] You have provided details below around the validation steps performed to gain confidence in these changes.

Using development data, I tested adding variables into `dbt_project.yml` that returned the proper account ids into each double entry model. If the value didn't exist, it would return empty models since the filter would exclude all data. The primary model tested was `int_quickbooks__bill_payment_double_entry` since it contained sufficient development data to test multiple field values and verify README instructions.

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR
    
- [x] README updates have been applied (if applicable)
    
- [NA] DECISIONLOG updates have been updated (if applicable)
- [NA] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [x] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
🪠

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,6 +128,19 @@ vars:
     quickbooks_union_databases: ['quickbooks_usa','quickbooks_canada'] # use this if the data is in different databases/projects but uses the same schema name
 ``` 
 
+### Configuring Account Type Names
+QuickBooks does allow for the `account_type` and `account_subtype` field values to be configured for customers internally. While our current filtered logic within our intermediate models accounts for the default values, it's possible there are different or multiple values that might change what the account types would be. 
+
+If you have different field value filters you'd like to set, you will need to configure the `account_type` and `account_subtype` variables that account for these variables in your `dbt_project.yml`.
```
Comment (fivetran-joemarkiewicz) [inline README.md:134]: This is a bit misleading. We do reference account_type and account_subtype in multiple locations throughout the package, but this update specifically handles cases where we are referencing a specific account_type/subtype to be used in the double entry models. I would rephrase this to specifically call out this scenario. Something along the following (which you partly cover in the yml block below):

```suggestion
Within a few of the double entry models in this package a mapping takes place to assign certain transaction type's debits/credits to the appropriate offset account (ie. Accounts Payable, Accounts Receivable, Undeposited Funds, and SalesOfProductIncome) reference. While our current filtered logic within our intermediate models account for the default values, it's possible your use case relies on different account types to reference. 

If you have a different value to reference for each type, you will need to configure the `account_type` and `account_sub_type` variables that account for these variables in your `dbt_project.yml`.
```

File: README.md [hunk 2]
```diff
@@ -128,6 +128,19 @@ vars:
...
+
+```yml
+vars: 
+  quickbooks__accounts_payable_reference: accounts_payable_value # 'Accounts Payable' is the default filter set
+  quickbooks__accounts_receivable_reference: account_receivable_value # 'Accounts Receivable' is the default filter set
+  quickbooks__undeposited_funds_reference: account_undeposited_funds_value # 'UndepositedFunds' is the default filter set
+  quickbooks__sales_of_product_income_reference: account_sales_of_product_income_value # 'SalesOfProductIncome' is the default filter set
```
Comment (fivetran-joemarkiewicz) [inline README.md:141]: I would specify with each if it is either on the `account_type` or `account_sub_type`

```suggestion
vars: 
  quickbooks__accounts_payable_reference: accounts_payable_value # 'Accounts Payable' is the default filter set for the account_type reference.
  quickbooks__accounts_receivable_reference: account_receivable_value # 'Accounts Receivable' is the default filter set for the account_type reference.
  quickbooks__undeposited_funds_reference: account_undeposited_funds_value # 'UndepositedFunds' is the default filter set for the account_sub_type reference.
  quickbooks__sales_of_product_income_reference: account_sales_of_product_income_value # 'SalesOfProductIncome' is the default filter set for the account_sub_type reference.
```

File: README.md [hunk 3]
```diff
@@ -202,14 +215,14 @@ vars:
 ``` 
 </details>
 
-## (Optional) Step 6: Orchestrate your models with Fivetran Transformations for dbt Core™
+## (Optional) Step 7: Orchestrate your models with Fivetran Transformations for dbt Core™
```
Comment (fivetran-joemarkiewicz) [inline README.md:218]: I don't believe this is needed anymore and can be `6` as we are no longer making a separate step for these variables.

File: README.md [hunk 4]
```diff
@@ -202,14 +215,14 @@ vars:
...
 <details><summary>Expand for details</summary>
 <br>
 
 Fivetran offers the ability for you to orchestrate your dbt project through [Fivetran Transformations for dbt Core™](https://fivetran.com/docs/transformations/dbt). Learn how to set up your project for orchestration through Fivetran in our [Transformations for dbt Core setup guides](https://fivetran.com/docs/transformations/dbt#setupguide).
 </details>
 
-## (Optional) Step 7: Validate your data
+## (Optional) Step 8: Validate your data
```
Comment (fivetran-joemarkiewicz) [inline README.md:225]: I don't believe this is needed anymore and can be `7` as we are no longer making a separate step for these variables.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-avinash thanks for working through this update! I only have a few requests to update before this will be good to go. Let me know if you have any questions on my suggested documentation changes.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,19 @@
+# dbt_quickbooks v0.11.1 
+[PR #98](https://github.com/fivetran/dbt_quickbooks/pull/98) includes the following updates:
+
+## 🐛 Bug Fixes 🩹
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply CHANGELOG.md]: Updated!

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,19 @@
...
+| [int_quickbooks__credit_memo_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__credit_memo_double_entry) | `quickbooks__accounts_receivable_reference` |
+| [int_quickbooks__deposit_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__deposit_double_entry) | `quickbooks__undeposited_funds_reference` |
+| [int_quickbooks__invoice_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__invoice_double_entry) | `quickbooks__sales_of_product_income_reference`, `quickbooks__accounts_receivable_reference` |
+| [int_quickbooks__payment_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__payment_double_entry) | `quickbooks__accounts_receivable_reference` |
+
+- Updated README with additional steps for configuration. This is an optional step since most customers will rely on the default account type/subtype values available.
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply CHANGELOG.md:17]: Done!

File: README.md (hunk 1 above)
Comment (fivetran-avinash) [reply README.md]: Updated!

File: README.md (hunk 2 above)
Comment (fivetran-avinash) [reply README.md]: Updated!

File: README.md (hunk 3 above)
Comment (fivetran-avinash) [reply README.md]: Oops, fixed now.

File: README.md (hunk 4 above)
Comment (fivetran-avinash) [reply README.md]: Fixed.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,21 @@
...
+
+## 🐛 Bug Fixes 🩹
+| **Updated model** | **New variables to filter on** |
+| ----------| -------------------- |
+| [int_quickbooks__bill_payment_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__bill_payment_double_entry) | `quickbooks__accounts_payable_reference` |
+| [int_quickbooks__credit_memo_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__credit_memo_double_entry) | `quickbooks__accounts_receivable_reference` |
+| [int_quickbooks__deposit_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__deposit_double_entry) | `quickbooks__undeposited_funds_reference` |
+| [int_quickbooks__invoice_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__invoice_double_entry) | `quickbooks__sales_of_product_income_reference`, `quickbooks__accounts_receivable_reference` |
+| [int_quickbooks__payment_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__payment_double_entry) | `quickbooks__accounts_receivable_reference` |
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:14]: I would actually consider the new variables as the feature and then what how they may be configured as the bugfix. See my suggestion below.

```suggestion
## 🐛 Bug Fixes 🩹 
- Updated intermediate double entry models that have `account_type` and `account_sub_type` filters with configurable variables, since the type names used in the filter may be different than the defaults for QuickBooks customers.

## 🎉 Feature Updates 🎉
- The variables to define the `account_type` and `account_sub_type` references were introduced in the following intermediate models: 
| **Updated model** | **New variables to filter on** |
| ----------| -------------------- |
| [int_quickbooks__bill_payment_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__bill_payment_double_entry) | `quickbooks__accounts_payable_reference` |
| [int_quickbooks__credit_memo_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__credit_memo_double_entry) | `quickbooks__accounts_receivable_reference` |
| [int_quickbooks__deposit_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__deposit_double_entry) | `quickbooks__undeposited_funds_reference` |
| [int_quickbooks__invoice_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__invoice_double_entry) | `quickbooks__sales_of_product_income_reference`, `quickbooks__accounts_receivable_reference` |
| [int_quickbooks__payment_double_entry](https://fivetran.github.io/dbt_quickbooks/#!/model/model.quickbooks.int_quickbooks__payment_double_entry) | `quickbooks__accounts_receivable_reference` |
```

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-avinash these look good! I just have a small suggestion to reword the CHANGELOG. Once that is applied, this will be good to go!

Actually can you hold off on merging just to discuss another change that I would like to include. We can discuss during standup. It has to do with PR #100
